hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -1004,6 +1005,89 @@ public AbfsRestOperation checkAccess(String path, String rwx, TracingContext tra
     return op;
   }
 
+  /**
+   * @return the properties returned from server.
+   * @throws AzureBlobFileSystemException in case it is not a 404 error or some other exception
+   * which was not able to be retried.
+   * */
+  public AbfsRestOperation getBlobProperty(Path blobPath,
+      TracingContext tracingContext) throws AzureBlobFileSystemException {
+    AbfsUriQueryBuilder abfsUriQueryBuilder = createDefaultUriQueryBuilder();
+    String blobRelativePath = blobPath.toUri().getPath();
```

Comment:
Is test for root path covered ?

Reply:
This method would fail in case of root level check with 400. For container, we would have to make GetContainerProperties API (https://learn.microsoft.com/en-us/rest/api/storageservices/get-container-properties?tabs=azure-ad). 
This would be required to do when we translate getFileStatus for Blob endpoint.

For rename, test has been added for testing the scenario where dest is root. ref: https://github.com/saxenapranav/hadoop/blob/5feeeb827e5800353b9a9cd4e96ec2f8d96289ae/hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemRename.java#L192-L198

Reply:
Yes, for src, it is already there: https://github.com/apache/hadoop/blob/5feeeb827e5800353b9a9cd4e96ec2f8d96289ae/hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemRename.java#L166-L177.

For dst, the new one:
https://github.com/apache/hadoop/blob/5feeeb827e5800353b9a9cd4e96ec2f8d96289ae/hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemRename.java#L192-L198

Reply:
ok, so there was pre-existing check that would have stopped src as root for  DFS too.

Reply:
for rename  in which src is root: on both dfs and for blob, it will fail. it will not go to server.
There is an existing code in AzureBlobFileSystem, where it check if `src.getParent != null`. if its null, it returns false.

for rename in which dst is root: on both dfs and for blob, it would pass and it will go to the server.